### ISSUE TITLE
E402: allow putting an import after conditional import

### ISSUE THREAD

**Issue by jack1142** [charliermarsh/ruff#151]
Example reproduction code:
```py
try:
    import tomllib
except ModuleNotFoundError:
    CAN_PARSE_TOML = False
else:
    CAN_PARSE_TOML = True

import sys

if CAN_PARSE_TOML:
    import os

import asyncio

asyncio, os, tomllib, sys  # silences F401 since it's not relevant
```
Output:
```py
❯ ruff repro.py -v
[2022-09-11][05:56:11][ruff::pyproject][DEBUG] Found pyproject.toml at: pyproject.toml
[2022-09-11][05:56:11][ruff][DEBUG] Identified files to lint in: 15.462µs
[2022-09-11][05:56:11][ruff][DEBUG] Checked files in: 1.800791ms
repro.py:8:1: E402 Module level import not at top of file
repro.py:13:1: E402 Module level import not at top of file

Found 2 error(s).

❯ flake8 repro.py -v        
flake8.checker            MainProcess     84 INFO     Making checkers
flake8.checker            MainProcess     85 INFO     Checking 1 files
flake8.main.application   MainProcess     90 INFO     Finished running
flake8.main.application   MainProcess     90 INFO     Reporting errors
flake8.main.application   MainProcess     90 INFO     Found a total of 0 violations and reported 0
```

I believe flake8 allows any code in try and if blocks without causing E402. It might also not be limited to try and if, not sure on that.

**Comment by charliermarsh** [comment]
Yeah this is supposed to be supported but there must be a bug. Thanks :)
